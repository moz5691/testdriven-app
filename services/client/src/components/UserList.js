import React from 'react';

const UserList = (props) => (
  <div>
    {
      props.users.map((user) => (
        <h4 key={user.id} className='box title is-4'>
            {user.username}
        </h4>
      ))
    }
  </div>
)

export default UserList;