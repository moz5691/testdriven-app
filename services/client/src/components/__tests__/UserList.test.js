import {shallow} from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import UserList from '../UserList';

const users = [
  {
    'active': true,
    'email': 'michael@gmail.com',
    'id': 1,
    'username': 'michael',
  },
  {
    'active': true,
    'email': 'jonnykay@gmail.com',
    'id': 2,
    'username': 'jonny1',
  }
];

test('UserList renders properly', () => {
  const wrapper =
      shallow(
          <UserList users =
           {
             users
           } />);
  const element = wrapper.find('h4');
  expect(element.length).toBe(2);
  expect(element.get(0).props.children).toBe('michael');
});

test('userList renders a snapshot properly', ()=>{
  const tree = renderer.create(<UserList users={users}/>)
          .toJSON();
  expect(tree).toMatchSnapshot();
});