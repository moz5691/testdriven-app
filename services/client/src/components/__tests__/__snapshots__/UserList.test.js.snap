// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userList renders a snapshot properly 1`] = `
<div>
  <h4
    className="box title is-4"
  >
    michael
  </h4>
  <h4
    className="box title is-4"
  >
    jonny1
  </h4>
</div>
`;
